Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp

```diff
@@ -145,27 +157,87 @@ ActsExamples::ProcessCode ActsExamples::TrackFindingAlgorithm::execute(
     // Clear trackContainerTemp and trackStateContainerTemp
     tracksTemp.clear();
 
-    auto result =
-        (*m_cfg.findTracks)(initialParameters.at(iseed), options, tracksTemp);
+    auto firstResult = (*m_cfg.findTracks)(initialParameters.at(iseed),
+                                           firstOptions, tracksTemp);
     m_nTotalSeeds++;
     nSeed++;
 
-    if (!result.ok()) {
+    if (!firstResult.ok()) {
       m_nFailedSeeds++;
       ACTS_WARNING("Track finding failed for seed " << iseed << " with error"
-                                                    << result.error());
+                                                    << firstResult.error());
       continue;
     }
 
-    auto& tracksForSeed = result.value();
-    for (auto& track : tracksForSeed) {
+    auto& firstTracksForSeed = firstResult.value();
+    for (auto& firstTrack : firstTracksForSeed) {
+      std::size_t nsecond = 0;
+
       // Set the seed number, this number decrease by 1 since the seed number
       // has already been updated
-      seedNumber(track) = nSeed - 1;
-      if (!m_trackSelector.has_value() ||
-          m_trackSelector->isValidTrack(track)) {
-        auto destProxy = tracks.getTrack(tracks.addTrack());
-        destProxy.copyFrom(track, true);  // make sure we copy track states!
+      seedNumber(firstTrack) = nSeed - 1;
+
+      if (m_cfg.twoWay) {
+        std::optional<Acts::VectorMultiTrajectory::TrackStateProxy> firstState;
+        for (auto st : firstTrack.trackStatesReversed()) {
+          bool isMeasurement =
+              st.typeFlags().test(Acts::TrackStateFlag::MeasurementFlag);
+          bool isOutlier =
+              st.typeFlags().test(Acts::TrackStateFlag::OutlierFlag);
+          // We are excluding non measurement states and outlier here. Those can
+          // decrease resolution because only the smoothing corrected the very
+          // first prediction as filtering is not possible.
+          if (isMeasurement && !isOutlier) {
+            firstState = st;
+          }
+        }
+
+        Acts::BoundTrackParameters secondInitialParameters(
+            firstState->referenceSurface().getSharedPtr(),
```

Comment:
if a `firstState` is not found it will crash here no?

Reply:
good point. I fixed it, pulled, fixed again, pulled, and then find you fixed it :)

Reply:
oh sorry 😅

Reply:
NP 😎